### PR TITLE
Feature/mdr_take_out_garbage/state_machine_definition

### PR DESCRIPTION
**Pull request by robertocaiwu**
This is a new scenario for the "take out the garbage" task for the German Open. The pull request includes the state machine definition for the task. [See #110] 

Note: The state machine is not completed yet. There are some packages that do not exist yet, and as such the arguments are left empty.

Corrections and suggestions are welcome.

### CODE REVIEW

**Comment by alex-mitrevski** [comment]
@robertocaiwu Do you think this is ready to be merged? If not, could you please add `WIP` to the PR title to indicate that you're still working on it?

**Comment by robertocaiwu** [comment]
I added the states I considered were missing on the SM. It is ready to be merged now.

File: mdr_planning/mdr_scenarios/mdr_robocup_tasks/mdr_take_out_garbage/config/take_out_garbage_sm.yaml [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+          - transition:
+              name: received_command
+              state: PROCESS_SPEECH_COMMAND
+    - state:
+        name: PROCESS_SPEECH_COMMAND
+        state_module_name: mdr_process_speech_command_action.action_states
+        state_class_name: ProcessCommand
+        transitions:
+          - transition:
+              name: start_store_groceries
```
Comment (alex-mitrevski) [inline mdr_planning/mdr_scenarios/mdr_robocup_tasks/mdr_take_out_garbage/config/take_out_garbage_sm.yaml]: Copy-paste error - should be `start_take_out_garbage` or something similar...

File: mdr_planning/mdr_scenarios/mdr_robocup_tasks/mdr_take_out_garbage/config/take_out_garbage_sm.yaml [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+                name: number_of_retries
+                value: 3
+    - state:
+        name: PERCEIVE_INSIDE_BIN
+        state_module_name: mdr_perception_behaviours.perceive_inside_bin
+        state_class_name: PerceiveInsideBin
+        transitions:
+            - transition:
+                name: succeeded
+                state: PICK_OBJECT
```
Comment (alex-mitrevski) [inline mdr_planning/mdr_scenarios/mdr_robocup_tasks/mdr_take_out_garbage/config/take_out_garbage_sm.yaml]: I think it's better to call this state `PICK_GARBAGE_BAG`.

File: mdr_planning/mdr_scenarios/mdr_robocup_tasks/mdr_take_out_garbage/config/take_out_garbage_sm.yaml [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+            - transition:
+                name: failed
+                state: GO_TO_COLLECTION_ZONE
+            - transition:
+                name: failed_after_retrying
+                state: FAILED
+        arguments:
+            - argument:
+                name: destination_locations
+                value: [COLLETION_ZONE]
```
Comment (alex-mitrevski) [inline mdr_planning/mdr_scenarios/mdr_robocup_tasks/mdr_take_out_garbage/config/take_out_garbage_sm.yaml]: COLLE**C**TION

File: mdr_planning/mdr_scenarios/mdr_robocup_tasks/mdr_take_out_garbage/config/take_out_garbage_sm.yaml [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+                name: plane_prefix
+                value: collection_zone
+    - state:
+        name: PLACE_OBJECT
+        state_module_name: mdr_manipulation_behaviours.place
+        state_class_name: Place
+        transitions:
+            - transition:
+                name: succeeded
+                state: CHECK_EMPTY_TABLE
```
Comment (alex-mitrevski) [inline mdr_planning/mdr_scenarios/mdr_robocup_tasks/mdr_take_out_garbage/config/take_out_garbage_sm.yaml]: `CHECK_EMPTY_TABLE` -> `CHECK_BINS_LEFT`

**Review by alex-mitrevski — CHANGES_REQUESTED** [review]

**Comment by argenos** [comment]
I just merged #125, for Travis checks to pass, you need the latest changes.

**Review by alex-mitrevski — APPROVED** [review]

**Comment by argenos** [comment]
The build is failing because the python package is not properly configured, can you take a look?

**Comment by robertocaiwu** [comment]
I believe this is ready to be merged.

**Pull request merged**